palace/drivers/basesolver.cpp

```diff
@@ -57,53 +58,20 @@ void WriteMetadata(const std::string &post_dir, const json &meta)
   fo << meta.dump(2) << '\n';
 }
 
-}  // namespace
-
-BaseSolver::BaseSolver(const IoData &iodata, bool root, int size, int num_thread,
-                       const char *git_tag)
-  : iodata(iodata), post_dir(GetPostDir(iodata.problem.output)), root(root), table(8, 9, 9)
-{
-  // Create directory for output.
-  if (root && !std::filesystem::exists(post_dir))
-  {
-    std::filesystem::create_directories(post_dir);
-  }
-
-  // Initialize simulation metadata for this simulation.
-  if (root && post_dir.length() > 0)
-  {
-    json meta;
-    if (git_tag)
-    {
-      meta["GitTag"] = std::string(git_tag);
-    }
-    if (size > 0)
-    {
-      meta["Problem"]["MPISize"] = size;
-    }
-    if (num_thread > 0)
-    {
-      meta["Problem"]["OpenMPThreads"] = num_thread;
-    }
-    WriteMetadata(post_dir, meta);
-  }
-}
-
-std::string BaseSolver::IterationPostDir(int iter) const
+// Creates a directory for storing postprocessing data for a given adaptation
+// iteration, and returns a path
+fs::path IterationPostDir(bool root, const std::string &post_dir, int iter)
 {
   std::string dir = fmt::format("{}Adapt{:0>3d}/", post_dir, iter);
   // Create directory for output.
-  if (root && !std::filesystem::exists(dir))
+  if (!fs::exists(dir))
   {
-    std::filesystem::create_directories(dir);
+    fs::create_directories(dir);
   }
-  return dir;
+  return fs::path(dir);
```

Comment:
Is there an advantage to `fs::path` instead of just `string` as used everywhere else?

Reply:
For this function it could be a string, the one place it gets called, in the `SavePostProcess` lambda, it would then be converted to a path immediately. Using `fs::path` allows for using the `directory_iterator` and `copy` within the lambda, in addition to the `/` operator. So this function could return a string, but `fs::path` would get called on the result immediately, and given the function is only used in this one place, it seemed neater to have it return the finished product. Either is fine with me though.